src/processing/queryprocessor.h

```diff
@@ -40,24 +40,22 @@ class QueryProcessor {
 				std::string getPlot();
 				ProvenanceCollection& getProvenance();
 				bool isError();
-				std::string getErrorMessage();
-				MappingExceptionType getErrorType();
+				MappingException getErrorException();
 
 			static std::unique_ptr<QueryResult> raster(std::unique_ptr<GenericRaster> result, const QueryRectangle &qrect, std::unique_ptr<ProvenanceCollection> provenance);
 				static std::unique_ptr<QueryResult> points(std::unique_ptr<PointCollection> result, const QueryRectangle &qrect, std::unique_ptr<ProvenanceCollection> provenance);
 				static std::unique_ptr<QueryResult> lines(std::unique_ptr<LineCollection> result, const QueryRectangle &qrect, std::unique_ptr<ProvenanceCollection> provenance);
 				static std::unique_ptr<QueryResult> polygons(std::unique_ptr<PolygonCollection> result, const QueryRectangle &qrect, std::unique_ptr<ProvenanceCollection> provenance);
 				static std::unique_ptr<QueryResult> plot(const std::string &plot, const QueryRectangle &qrect, std::unique_ptr<ProvenanceCollection> provenance);
-				static std::unique_ptr<QueryResult> error(const std::string &error, const QueryRectangle &qrect, const MappingExceptionType errorType);
+				static std::unique_ptr<QueryResult> error(const MappingException &exception, const QueryRectangle &qrect);
 			private:
-				QueryResult(Query::ResultType result_type, std::unique_ptr<SpatioTemporalResult> result, std::unique_ptr<ProvenanceCollection> provenance, const std::string &result_plot, const std::string &error, const QueryRectangle &qrect, MappingExceptionType errorType = MappingExceptionType::CONFIDENTIAL);
+				QueryResult(Query::ResultType result_type, std::unique_ptr<SpatioTemporalResult> result, std::unique_ptr<ProvenanceCollection> provenance, const std::string &result_plot, const MappingException &exception, const QueryRectangle &qrect);
 				Query::ResultType result_type;
 				std::unique_ptr<SpatioTemporalResult> result;
 				std::unique_ptr<ProvenanceCollection> provenance;
 				std::string result_plot;
-				std::string result_error;
 				QueryRectangle qrect;
-				MappingExceptionType errorType;
+				MappingException result_exception;
```

Comment:
What if there is no exception? Would it makes sense to wrap it with http://en.cppreference.com/w/cpp/utility/optional ?

Reply:
Yes, thats a good idea. But I used boost::optional, because std::optional is C++17.